File: BE_QuanLyTro/src/main/java/ut/edu/be_quanlytro/Dto/Response/ContractTerminationResponse.java
package ut.edu.be_quanlytro.Dto.Response;

import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;
import java.util.UUID;

@Getter @Setter @Builder
public class ContractTerminationResponse {
    private UUID contractId;
    private String roomNumber;
    private BigDecimal depositAmount;
    private BigDecimal totalDeduction;
    private BigDecimal finalAmount;
    private String settlementAction;
    private String message;
}

File: Flutter_QuanLyTro/lib/data/models/response/contract_termination_response.dart
class ContractTerminationResponse {
  final String contractId;
  final String roomNumber;
  final double depositAmount;
  final double totalDeduction;
  final double finalAmount;
  final String settlementAction;
  final String message;

  ContractTerminationResponse({
    required this.contractId,
    required this.roomNumber,
    required this.depositAmount,
    required this.totalDeduction,
    required this.finalAmount,
    required this.settlementAction,
    required this.message,
  });

  factory ContractTerminationResponse.fromJson(Map<String, dynamic> json) {
    return ContractTerminationResponse(
      contractId: json['contractId'] ?? '',
      roomNumber: json['roomNumber'] ?? '',
      depositAmount: (json['depositAmount'] ?? 0).toDouble(),
      totalDeduction: (json['totalDeduction'] ?? 0).toDouble(),
      finalAmount: (json['finalAmount'] ?? 0).toDouble(),
      settlementAction: json['settlementAction'] ?? '',
      message: json['message'] ?? '',
    );
  }
}